### PR TITLE
Adding a symlink to a file that is created with a trailing slash breaks walk-sync.

### PR DESCRIPTION
**Pull request by lukemelia**
This addition to the test suite illustrates the issue. (Note that there are no tests added, so this PR is really just a way to reproduce the problem.)

I ran into this trying to run the ember-native-class-codemod with the error captured here: https://github.com/ember-codemods/ember-native-class-codemod/issues/138

### CODE REVIEW

**Comment by stefanpenner** [comment]
Interesting, this seems to just break the test setup.

**Comment by lukemelia** [comment]
@stefanpenner yes, I think I've recreated the real-world error now.

**Comment by stefanpenner** [comment]
@lukemelia is the expected behavior surface the existence of the symlink, but not to attempt to traverse the broken symlink?


```
 s/j/node-walk-sync ╍ {pr/44} cd test/fixtures/symlink3
cd: The directory 'test/fixtures/symlink3' does not exist
```

**Comment by lukemelia** [comment]
I would say that node-walk-sync should warn but continue its traversal.

**Comment by stefanpenner** [comment]
@lukemelia I've pushed a potential fix.

File: index.ts [hunk 1]
```diff
@@ -27,9 +27,10 @@ function getStat(path: string) {
   try {
     return fs.statSync(path);
   } catch(error) {
-    if (error.code !== 'ENOENT') {
-      throw error;
+    if (error !== null && typeof error === 'object' && (error.code === 'ENOENT' || error.code === 'ENOTDIR')) {
```
Comment (stefanpenner) [inline index.ts]: @lukemelia potential fix.

We don't currently warn on broken symlinks, as other dir executable libraries such as `tree`  do not either.

Comment (lukemelia) [reply index.ts]: @stefanpenner got it. This seems like a reasonable approach then.

**Comment by stefanpenner** [comment]
released as v1.1.4 🎉

**Comment by lukemelia** [comment]
@stefanpenner do tests pass on master for you locally? I would have expected this PR to need some test changes.

**Comment by stefanpenner** [comment]
@lukemelia. Yup they do, and I made the needed changes (I believe)

**Comment by lukemelia** [comment]
Excellent, and ci too, so that's good enough for me! :-D

**Comment by stefanpenner** [comment]
@lukemelia thanks for reporting, let me know if you see anything funky!

**Pull request merged**